src/main/java/programminglife/parser/GraphParser.java

```diff
@@ -0,0 +1,177 @@
+package programminglife.parser;
+
+import com.diffplug.common.base.Errors;
+import com.diffplug.common.base.Throwing;
+import programminglife.model.Graph;
+import programminglife.model.Node;
+import programminglife.model.exception.UnknownTypeException;
+
+import java.io.*;
+import java.util.NoSuchElementException;
+import java.util.Observable;
+
+/**
+ * Created by toinehartman on 15/05/2017.
+ */
+public class GraphParser extends Observable implements Runnable {
+
+    private static final boolean PARSE_LINE_VERBOSE_DEFAULT = true;
+
+    private Graph graph;
+    private File graphFile;
+    private boolean verbose;
+
+    /**
+     * Initiates an empty graph and the {@link File} to parse.
+     * @param graphFile the file to parse the {@link Graph} from
+     */
+    public GraphParser(File graphFile) {
+        this.graphFile = graphFile;
+        this.verbose = PARSE_LINE_VERBOSE_DEFAULT;
+        this.graph = new Graph("");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void run() {
+        try {
+            System.out.printf("%s Parsing Graph on separate Thread", Thread.currentThread());
+            parse(this.verbose);
+            this.setChanged();
+            this.notifyObservers(this.graph);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (UnknownTypeException e) {
+            e.printStackTrace();
+        }
```

Comment:
Please actually do something when an exception is thrown (Notify observers that parsing failed?)